common/shared.c

```diff
@@ -85,6 +87,41 @@ const unsigned int test_cnt = (sizeof test_size / sizeof test_size[0]);
 static const char integ_alphabet[] = "0123456789abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ";
 static const int integ_alphabet_length = (sizeof(integ_alphabet)/sizeof(*integ_alphabet)) - 1;
 
+
+static void ft_close_fids(void)
+{
+	FT_CLOSE_FID(mr);
+	FT_CLOSE_FID(ep);
+	FT_CLOSE_FID(pep);
+	FT_CLOSE_FID(rxcq);
+	FT_CLOSE_FID(txcq);
+	FT_CLOSE_FID(rxcntr);
+	FT_CLOSE_FID(txcntr);
+	FT_CLOSE_FID(av);
+	FT_CLOSE_FID(eq);
+	FT_CLOSE_FID(pollset);
+	FT_CLOSE_FID(domain);
+	FT_CLOSE_FID(waitset);
+	FT_CLOSE_FID(fabric);
+}
+
+void ft_free_res(void)
+{
+	ft_close_fids();
+	if (buf) {
```

Comment:
Yeah, but free() is just going to do it anyway, and it saves an unnecessary memory write.